R/utils.R

```diff
@@ -11,4 +11,33 @@ weighted_mean <- function(x, w, ..., na.rm = FALSE){
     x <- x[keep]
   }
   weighted.mean(x, w, ..., na.rm = FALSE)
+}
+
+#' Function to split species traits data.frame into a list based on provided
+#' species categories
+#' 
+#' @noRd
+split_species_categories <- function(
```

Comment:
Maybe it could be good to check for missing species in one of the two objects and to return an error/warning/message. The reasons can be: mispelled species, missing species, or species out of the category.

Reply:
Hum... For now I'm only merging species categories and their traits, but for sure, we could output a message in case of a difference in number of provided species

Reply:
I have opened an issue. We can see later if this is pertinent or not